rdr_service/alembic/versions/7c0585790bc8_update_sp_get_code_module.py

```diff
@@ -0,0 +1,188 @@
+down_revision = '377e8ced743f'
+branch_labels = None
+depends_on = None
+
+sp_get_code_module_items = ReplaceableObject(
+    "sp_get_code_module_items",
+    """
+  (IN module VARCHAR(80))
+  BEGIN
+    # Return all of the codebook items (topics, questions, answers) related to the passed
+    # module name.
+    SELECT @code_id := code_id FROM code WHERE `value` = module and parent_id is NULL;
+
+    select code_id,
+       parent_id,
+       topic,
+       code_type,
+       `value`,
+       display,
+       `system`,
+       mapped,
+       created,
+       code_book_id,
+       short_value
+    from code where code_id in
+    (
+        select distinct s.code_id
+        from survey s
+        where s.code_id=@code_id
+        UNION
+        select distinct sq.code_id
+        from survey s, survey_question sq
+        where s.id=sq.survey_id
+        and s.code_id=@code_id
+        UNION
+        select distinct sqo.code_id
+        from survey s, survey_question sq, survey_question_option sqo
+        where s.id=sq.survey_id and sq.id=sqo.question_id
+        and s.code_id=@code_id
+        )
+    order by code_id;
+  END;
+""",
+)
+
+sp_get_questionnaire_answers = ReplaceableObject(
+    "sp_get_questionnaire_answers",
+    """
+  (IN module VARCHAR(80), IN participant_id INT)
+  BEGIN
+    # Dynamically pivot the questionnaire answers for the given participant and module.
+    # Results are ordered by 'created' descending.
+    DECLARE CONTINUE HANDLER FOR 1064 SET @sql = NULL;
+    DECLARE CONTINUE HANDLER FOR 1243 SELECT 1 AS 'invalid_code_id' FROM dual WHERE FALSE;
+
+    SET @sql = '';
+    SELECT @module := COALESCE(c.value, 0), @code_id := COALESCE(c.code_id, 0)
+    FROM code c
+    WHERE c.value = module;
+
+    SELECT @sql := CONCAT(@sql, IF(@sql = '', '', ', '), temp.output)
+    FROM (
+             SELECT DISTINCT CONCAT('GROUP_CONCAT(IF(code_id = ', code_id, ', answer, NULL) SEPARATOR ",") AS ',
+                                    `value`) as output
+             FROM (
+                    select code_id, value, display, code_type, parent_id, '0' AS sort_id
+                    from code where code_id in
+                    (
+                        select distinct s.code_id
+                        from survey s
+                        where s.code_id=@code_id
+                        UNION
+                        select distinct sq.code_id
+                        from survey s, survey_question sq
+                        where s.id=sq.survey_id
+                        and s.code_id=@code_id
+                        UNION
+                        select distinct sqo.code_id
+                        from survey s, survey_question sq, survey_question_option sqo
+                        where s.id=sq.survey_id and sq.id=sqo.question_id
+                        and s.code_id=@code_id
+                    )
+                    order by code_id
+                  ) b
+             WHERE b.code_type = 3
```

Comment:
If this subquery is only looking for question codes, it may be better to look at just the codes from `survey_question` table (rather than getting all of the codes for a survey and filtering by the type on the code table. The `code_type` column isn't very reliable anymore either, since a given code can be reused as different types.

Reply:
@kenny-skaggs Can a code be reused as a different type, or just be reused as the same type but across different surveys?  Do you have an example of it being used as a different type?  (Afraid the PDR generators wouldn't handle that)

@wangy70 I'm trying to reconcile this logic with two places in the PDR-related code.  First, in the BQ models, when we try to find all the question codes associated with a module:
https://github.com/all-of-us/raw-data-repository/blob/810fd28ef2a5df2e45712edae4e36d520c105c15/rdr_service/model/bq_questionnaires.py#L70

...and then when we are looking for the answers from a specific response payload (where we will "roll up" answers from the previous response if the more recent response did not have that question/answer code in the partial payload):
https://github.com/all-of-us/raw-data-repository/blob/810fd28ef2a5df2e45712edae4e36d520c105c15/rdr_service/dao/bq_questionnaire_dao.py#L24

Reply:
@pegbertsch, it doesn't look like any codes are reused as different types yet. And maybe that's not a coincidence, but I know the possibility has been brought up before. It seems like it could be a good way of 'future proofing' if we already have the capability of recognizing them as questions through the `survey_question` table rather than reading from the `code_type` column.

Reply:
@kenny-skaggs Good catch, done!